File: nicolaspuebla_proyecto_final/src/main/java/com/example/nicolaspuebla_proyecto_final/model/dto/MatchCreation.java
package com.example.nicolaspuebla_proyecto_final.model.dto;

import com.example.nicolaspuebla_proyecto_final.model.dataModels.Team;

/**
 * Clase que contiene la información necesaria para la creación de un partido.
 */
public class MatchCreation extends EventCreation {
    private Team opponent;
    private int ownGoals;
    private int opponentGoals;

    public MatchCreation(){}

    public MatchCreation(String event_type, Long teamId, Double latitude, Double longitude, String date, Team opponent, int ownGoals, int opponentGoals) {
        super(event_type, teamId, latitude, longitude, date);
        this.opponent = opponent;
        this.ownGoals = ownGoals;
        this.opponentGoals = opponentGoals;
    }

    public Team getOpponent() {
        return opponent;
    }

    public void setOpponent(Team opponent) {
        this.opponent = opponent;
    }

    public int getOwnGoals() {
        return ownGoals;
    }

    public void setOwnGoals(int ownGoals) {
        this.ownGoals = ownGoals;
    }

    public int getOpponentGoals() {
        return opponentGoals;
    }

    public void setOpponentGoals(int opponentGoals) {
        this.opponentGoals = opponentGoals;
    }   
}

File: nicolaspuebla_proyecto_final_android/app/src/main/java/com/example/nicolaspuebla_proyecto_final_android/data/model/dto/MatchCreation.kt
package com.example.nicolaspuebla_proyecto_final_android.data.model.dto

import com.example.nicolaspuebla_proyecto_final_android.data.model.dataClases.Team
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Clase que se emplea en la petición de creación de un equipo.
 */
data class MatchCreation (
    override val event_type: String = "match",
    override val teamId: Long,
    override val latitude: Double,
    override val longitude: Double,
    override val date: String,
    val opponent: Team,
    @JsonProperty("own_goals")
    val ownGoals: Int,
    @JsonProperty("opponent_goals")
    val opponentGoals: Int
): EventCreation(event_type, teamId, latitude, longitude, date)